src/mongoc/mongoc-collection.c

```diff
@@ -1961,7 +1961,7 @@ mongoc_collection_find_and_modify (mongoc_collection_t *collection,
     */
    ret = _mongoc_client_command_simple_with_hint (collection->client,
                                                   collection->db, &command, NULL,
-                                                  reply, selected_server->id, error);
+                                                  true, reply, selected_server->id, error);
```

Comment:
This change wasn't necessary for the test suite to pass, since `test_find_and_modify()` in `test-mongoc-collection.c` performs no assertions on the request; however, I think it's logical that findAndModify be considered a write command in the context of this PR. Let me know if you disagree.